neo/Consensus/ConsensusService.cs

```diff
@@ -110,11 +110,14 @@ private void CheckCommits()
         private void CheckExpectedView(byte viewNumber)
         {
             if (context.ViewNumber == viewNumber) return;
-            if (context.ChangeViewPayloads.Count(p => p != null && p.GetDeserializedMessage<ChangeView>().NewViewNumber == viewNumber) >= context.M())
+            // if there are `M` change view payloads with NewViewNumber greater than viewNumber, then, it is safe to move
+            if (context.ChangeViewPayloads.Count(p => p != null && p.GetDeserializedMessage<ChangeView>().NewViewNumber >= viewNumber) >= context.M())
```

Comment:
They are not in the same view, and it's useless to move to that view.

Reply:
I know it is, @erikzhang.

Reply:
We need to allow nodes to move to a higher view, another option could be to store all history of changeview payloads received.

Reply:
There is no enough information in `RecoveryMessage` to recover `ChangeView`. Because now the `ViewNumber` are not the same as `RecoveryMessage`.

Reply:
Exactly.

Maybe we keep this useless changeview for 2.10.1 and improve this for the next version.

Let just fix NewVN to static for standarzing the Payload.

Reply:
You need to remove `OriginalViewNumber` from `RecoveryMessage.ChangeViewPayloadCompact` and add `NewViewNumber` to it. Then you can get `OriginalViewNumber` from `NewViewNumber - 1`.

Reply:
I think that we already have ViewNumber from the inherited class.
We can directly use it and the static `NewViewNumber` will always be `ViewNumber++`

Reply:
There is no `ViewNumber` in `RecoveryMessage.ChangeViewPayloadCompact`.

Reply:
The recovery message itself extends `ConsensusMessage` which has the view number of the node sending recovery; this should be the `NewViewNumber` or all the change view messages that the recovery message contains.

Reply:
When a node changes view (during reset) it should latch all the `ChangeView` Payloads it uses.

Reply:
> The recovery message itself extends ConsensusMessage which has the view number of the node sending recovery

`>=`

Reply:
@erikzhang,

I think that this current PR is ok like this.
The simplification that is possible and simple for now could be something like this: https://github.com/neo-project/neo/pull/670

After 2.10.1 we could simplify something more and optimize this changeview to a lower view.

However, this looks like to be the simplest way to do it right now without any other bigger change.

Reply:
If you allow `>=` here, the `RecoveryMessage` may stop working unless you change `RecoveryMessage.ChangeViewPayloadCompact`.

Reply:
https://github.com/neo-project/neo/blob/8da7c2eb6d8aa61fdc1ac6619dcd67955fb189a4/neo/Consensus/RecoveryMessage.cs#L50-L55